site_scons/components/__init__.py

```diff
@@ -173,9 +173,26 @@ def define_mercury(reqs):
                 libs=['opa'],
                 package='openpa-devel' if inst(reqs, 'openpa') else None)
 
+    ucx_configure = ['./configure', '--disable-assertions', '--disable-params-check', '--enable-mt',
+                     '--without-go', '--without-java', '--prefix=$UCX_PREFIX']
+
+    if reqs.target_type == 'debug':
+        ucx_configure.append('--enable-debug')
+    else:
+        ucx_configure.append('--disable-debug')
+
     reqs.define('ucx',
+                retriever=GitRepoRetriever('https://github.com/openucx/ucx.git'),
```

Comment:
I'm curious how this knows to use the UCX version in build.config?

Reply:
It's all built into the code in PreReqComponent in site_scons/prereq_tools/base.py.  It automatically loads the config file and looks for matching component.